src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Middleware/AddFifoStampTest.php

```diff
@@ -0,0 +1,127 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Tests\Middleware;
+
+use Symfony\Component\Messenger\Bridge\AmazonSqs\Middleware\AddFifoStamp;
+use Symfony\Component\Messenger\Bridge\AmazonSqs\Middleware\WithMessageDeduplicationId;
+use Symfony\Component\Messenger\Bridge\AmazonSqs\Middleware\WithMessageGroupId;
+use Symfony\Component\Messenger\Bridge\AmazonSqs\Transport\AmazonSqsFifoStamp;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Test\Middleware\MiddlewareTestCase;
+
+class AddFifoStampTest extends MiddlewareTestCase
+{
+    public function testAddStampWithGroupIdOnly()
+    {
+        $middleware = new AddFifoStamp();
+        $envelope = new Envelope(new WithMessageGroupIdMessage('groupId'));
+        $finalEnvelope = $middleware->handle($envelope, $this->getStackMock());
+        $stamp = $finalEnvelope->last(AmazonSqsFifoStamp::class);
+        $this->assertNotNull($stamp);
+        /* @var AmazonSqsFifoStamp $stamp */
+        $this->assertEquals('groupId', $stamp->getMessageGroupId());
+        $this->assertNull($stamp->getMessageDeduplicationId());
+    }
+
+    public function testHandleWithDeduplicationIdOnly()
+    {
+        $middleware = new AddFifoStamp();
+        $envelope = new Envelope(new WithMessageDeduplicationIdMessage('deduplicationId'));
+        $finalEnvelope = $middleware->handle($envelope, $this->getStackMock());
+        $stamp = $finalEnvelope->last(AmazonSqsFifoStamp::class);
+        $this->assertNotNull($stamp);
+        /* @var AmazonSqsFifoStamp $stamp */
+        $this->assertEquals('deduplicationId', $stamp->getMessageDeduplicationId());
+        $this->assertNull($stamp->getMessageGroupId());
+    }
+
+    public function testHandleWithGroupIdAndDeduplicationId()
+    {
+        $middleware = new AddFifoStamp();
+        $envelope = new Envelope(new WithMessageDeduplicationIdAndMessageGroupIdMessage('my_group', 'my_random_id'));
+        $finalEnvelope = $middleware->handle($envelope, $this->getStackMock());
+        $stamp = $finalEnvelope->last(AmazonSqsFifoStamp::class);
+        $this->assertNotNull($stamp);
+        /* @var AmazonSqsFifoStamp $stamp */
+        $this->assertEquals('my_random_id', $stamp->getMessageDeduplicationId());
+        $this->assertEquals('my_group', $stamp->getMessageGroupId());
+    }
+
+    public function testHandleWithoutId()
+    {
+        $middleware = new AddFifoStamp();
+        $envelope = new Envelope(new WithoutIdMessage());
+        $finalEnvelope = $middleware->handle($envelope, $this->getStackMock());
+        $stamp = $finalEnvelope->last(AmazonSqsFifoStamp::class);
+        /* @var AmazonSqsFifoStamp $stamp */
+        $this->assertNull($stamp);
+    }
+}
+
+class WithMessageDeduplicationIdAndMessageGroupIdMessage implements WithMessageDeduplicationId, WithMessageGroupId
```

Comment:
another approach might be to use anonymous classes instead of dummy classes

Reply:
yes indeed ! I find dummy more easy to understand in these case but a matter of taste I guess.

I can move to anonymous classes if it can help merge ;)

Reply:
No lets keep them, its fine 👍